src/bitwarden_workflow_linter/models/step.py

```diff
@@ -39,10 +39,13 @@ def init(cls: Self, idx: int, job: str, data: CommentedMap) -> Self:
         new_step.key = idx
         new_step.job = job
 
-        if "uses" in data.ca.items and data.ca.items["uses"][2]:
-            new_step.uses_comment = data.ca.items["uses"][2].value.replace("\n", "")
+        if new_step.uses:
+            if "uses" in data.ca.items and data.ca.items["uses"][2]:
+                new_step.uses_comment = data.ca.items["uses"][2].value.replace("\n", "")
+                new_step.uses_version = new_step.uses_comment.split(" ")[-1]
             if "@" in new_step.uses:
                 new_step.uses_path, new_step.uses_ref = new_step.uses.split("@")
```

Comment:
This should be moved to the lower `if "@"...` check for reused workflows referenced within the same repo like `./.github/workflows/_docker.yml`, they normally don't have versions

```suggestion
                new_step.uses_comment = data.ca.items["uses"][2].value.replace("\n", "")
            if "@" in new_step.uses:
                new_step.uses_path, new_step.uses_ref = new_step.uses.split("@")
                new_step.uses_version = new_step.uses_comment.split(" ")[-1]
```

Reply:
The current code should  (and appears to) handle that situation correctly, but the suggested modification causes issues since `new_step.uses_comment` is only conditionally set in the if above, so it might be `None`, which breaks on the `split`.

Just to be sure the current code works in all cases I'll add some tests in since it didn't look like we had any w/ some of those variations.

Reply:
Oh! gotcha, the `uses_version` needs the `uses`_comment`